src/main/java/com/google/cloud/tools/managedcloudsdk/install/Extractor.java

```diff
@@ -81,4 +82,10 @@ ExtractorProvider getExtractorProvider() {
   private void cleanUp(final Path target) throws IOException {
     MoreFiles.deleteRecursively(target, RecursiveDeleteOption.ALLOW_INSECURE);
   }
+
+  static boolean isTargetInsideDestination(Path target, Path destination) throws IOException {
```

Comment:
I realized that this is re-canonicalizing the destination for every tar/zip entry.  Extraction already takes some time.

Reply:
That's a good point. I'll see what I can do.

Reply:
So I've been adding a Javadoc after making this accept a canonicalized destination path like this, and now I feel I would just remove this method and embed the code directly. We are basically talking about the necessity of trading simplicity for optimization, so embedding the code is reasonable and actually better than having a util method with a weird Javadoc.

```java
  /**
   * Check if {@code target} is inside {@code canonicalDestination}
   *
   * @param target path to check if inside {@code canonicalDestination}
   * @param canonicalDestination canonicalized destination path; must be canonicalized beforehand
   *     for this method to work correctly (behavior undefined otherwise)
   */
  // For optimization reasons, this method does not check if "canonicalDestination" is already
  // canonicalized.
  static boolean isTargetInsideDestination(Path target, String canonicalDestination)
```

Reply:
Optimized performance. This removes small tests for the method, but we still have end-to-end tests with actual Zip-Slip samples.